storage/queue.go

```diff
@@ -111,6 +112,96 @@ type PeekMessageResponse struct {
 	MessageText    string `xml:"MessageText"`
 }
 
+// QueueMetadata represents user defined metadata and queue properties on
+// a specific queue.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dd179384.aspx
+type QueueMetadata struct {
+	ApproximateMessageCount int
+	userDefinedMetadata     map[string]string
```

Comment:
I think you should just export this map (maybe all of the headers, doesn't matter) to the user and they decide what to do with it. Therefore Get/Set below is obsolete. We don't need that many helpers. We're trying to deliberately remove client-side magic from the SDK.